silo/management/commands/stripMongoData.py

```diff
@@ -0,0 +1,112 @@
+from django.conf import settings
+import pymongo
+from bson.objectid import ObjectId
+import json
+import random
+from silo.models import LabelValueStore, Silo
+
+
+class Command(BaseCommand):
+    """
+    Usage: python manage.py update_to_0-9-2
+    """
+    def add_arguments(self, parser):
+        parser.add_argument("--write", action='store_true', dest='write')
+
+    def handle(self, *args, **options):
+
+        client  = pymongo.MongoClient(settings.MONGODB_URI)
+        db = client.get_database(settings.TOLATABLES_MONGODB_NAME)
+
+        counter = 0
+        foundSilos = set()
+        recordsUpdated = []
+        multiMod = []
+
+        # sampleRecords = db.label_value_store.find({"silo_id": {"$in": [175, 891, 996, 981, 1342, 1576]}})
+        # print sampleRecords.count()
+        # for r in sampleRecords:
+        #     print 's=', r['silo_id']
+        # import sys
+        # print 'done'
+        # sys.exit()
+
+        autoPushEnabled = list(Silo.objects.filter(reads__autopush_frequency__isnull=False).values_list('pk', flat=True))
+        problemSilos = {}
+        diffSamples = []
+        diffInclude = ['5731956873abe223358d3a74', '57a473d005e48c0fdd83cdb8']
+        rCountBefore = db.label_value_store.count()
+
+        for record in db.label_value_store.find({}):
+            for key in record:
+                try:
+                    newVal = record[key].strip()
+                except:
+                    continue
+                # print 'new', newVal
+                # print '|val=%s newval=%s' % (record[key], newVal)
+                if newVal != record[key]:
+                    # print all diffs to console and save some samples for later
+                    diff = 'silo %s: different new=|%s| old=|%s|, key=%s, _id=%s' % \
+                        (record['silo_id'],
+                        smart_str(newVal),
+                        smart_str(record[key]),
+                        smart_str(key),
+                        record['_id'])
+                    print diff
+                    if (random.random() > .99 and len(diffSamples) <=15) \
+                            or str(record['_id']) in diffInclude:
+                        diffSamples.append(diff)
+
+                    # There could be problems with the GSheet data if the auto-push
+                    # has already taken place.  This lets us highlight potential problems
+                    if record['silo_id'] in autoPushEnabled:
+                        try:
+                            problemSilos[record['silo_id']].append(key)
+                        except KeyError:
+                            problemSilos[record['silo_id']] = [key]
+                    foundSilos.add(record['silo_id'])
+
+                    # Only perform the update if the --write flag is thrown
+                    if options['write']:
+                        rid = ObjectId(record['_id'])
+                        result = db.label_value_store.update_one(
```

Comment:
I'm not sure how long it takes in total with the real db, but it looks to me pretty inefficient to do single writes for each key instead of more bulk writes to DB. If it's fast, then ignore this comment

Reply:
It ran through 517,441 records in less than a minute, so fast enough for us.